internal/services/datadog/datadog_monitors_resource.go

```diff
@@ -0,0 +1,493 @@
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceDatadogMonitor() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceDatadogMonitorCreate,
+		Read:   resourceDatadogMonitorRead,
+		Update: resourceDatadogMonitorUpdate,
+		Delete: resourceDatadogMonitorDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.DatadogMonitorID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.DatadogMonitorsName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"datadog_organization_properties": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"api_key": {
+							Type:      pluginsdk.TypeString,
+							Required:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"application_key": {
+							Type:      pluginsdk.TypeString,
+							Required:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"enterprise_app_id": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"linking_auth_code": {
+							Type:      pluginsdk.TypeString,
+							Optional:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"linking_client_id": {
+							Type:      pluginsdk.TypeString,
+							Optional:  true,
+							ForceNew:  true,
+							Sensitive: true,
+						},
+
+						"redirect_uri": {
+							Type:     pluginsdk.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"identity": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"type": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"SystemAssigned",
+								"UserAssigned",
+							}, false),
+						},
+
+						"principal_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"tenant_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"sku": {
```

Comment:
How about simplifying this to `sku_name`?

Reply:
In our swagger we have this schema, so should it be matched or can i make change here?

Reply:
Yes, the tf schema is not necessary to be exact the same with swagger schema.

Reply:
Incorporating in next commit.